optimade/validator/validator.py

```diff
@@ -483,32 +490,39 @@ def _construct_queries_for_property(
                 f"Chosen entry of endpoint '/{endp}' had unexpected type '{chosen_entry['type']}'."
             )
 
-        if prop == "id":
-            prop_type = DataType.STRING
+        if prop_type is None:
+            if prop in CONF.entry_schemas[endp]:
+                prop_type = CONF.entry_schemas[endp].get(prop, {}).get("type")
 
         if prop_type is None:
             raise ResponseError(
-                f"Cannot validate queries on {prop} as field type was not reported in /info/{endp}"
+                f"Cannot validate queries on {prop} as field type was not reported in `/info/{endp}`"
             )
 
         # this is the case of a provider field
-        if prop not in (
-            CONF.entry_properties[endp]["MUST"]
-            | CONF.entry_properties[endp]["SHOULD"]
-            | CONF.entry_properties[endp]["OPTIONAL"]
-        ):
+        if prop not in CONF.entry_schemas[endp]:
             if self.provider_prefix is None:
                 raise ResponseError(
-                    f"Found unknown field {prop} and no provider prefix was provided in `/info`"
+                    f"Found unknown field '{prop}' in `/info/{endp}` and no provider prefix was provided in `/info`"
                 )
             elif not prop.startswith(f"_{self.provider_prefix}"):
                 raise ResponseError(
-                    f'Found unknown field {prop} that did not start with provider prefix "_{self.provider_prefix}"'
+                    f"Found unknown field '{prop}' that did not start with provider prefix '_{self.provider_prefix}'"
                 )
-            return True, f"Found provider field {prop}, will not test queries."
+            return (
+                True,
+                f"Found provider field '{prop}', will not test queries as they are strictly optional.",
+            )
+
+        query_optional = False
+        if prop in CONF.entry_schemas[endp]:
+            query_optional = (
+                CONF.entry_schemas[endp].get(prop, {}).get("queryable")
+                == SupportLevel.OPTIONAL
+            )
 
         return self._construct_single_property_filters(
-            prop, prop_type, sortable, endp, chosen_entry
+            prop, prop_type, sortable, endp, chosen_entry, query_optional=query_optional
```

Comment:
I would suggest you to either do all with keywords or non, for this addition alone, I would hence follow the existing trend:
```suggestion
            prop, prop_type, sortable, endp, chosen_entry, query_optional
```